### PR TITLE
PM-11142 - Implement empty vault screen UI

### PR DESCRIPTION
**Pull request by phil-livefront**
## 🎟️ Tracking

[PM-11142](https://bitwarden.atlassian.net/browse/PM-11142)

## 📔 Objective

- Update the empty vault list to match the new designs for the new reskin designs

## 📸 Screenshots

Simulator Screenshot - iPhone 15 Pro - 2024-10-08 at 09 20 40

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11142]: https://bitwarden.atlassian.net/browse/PM-11142?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Vault/Vault/VaultList/VaultListView.swift [hunk 1]
```diff
@@ -69,30 +69,44 @@ private struct SearchableVaultListView: View {
...
+            vaultFilterRow
+                .padding(.top, 16)
+
+            Spacer()
+
+            Image(decorative: Asset.Images.emptyVaultItems)
+                .resizable()
+                .frame(width: 100, height: 100)
+
+            VStack(spacing: 12) {
```
Comment (matt-livefront) [inline BitwardenShared/UI/Vault/Vault/VaultList/VaultListView.swift:82]: 🎨 It looks like this `VStack` should switch to an `HStack` in landscape, should we do that as part of this? Would it work to use `PageHeaderView` here?

https://www.figma.com/design/eH8ybfmeq726N1SyvcoK2K/Bitwarden-iOS-Mobile-App-Reskin---V3-(current)?node-id=13870-311465&t=8HWibq0igAUVoBqk-4

File: BitwardenShared/UI/Vault/Vault/VaultList/VaultListView.swift [hunk 2]
```diff
@@ -69,30 +69,44 @@ private struct SearchableVaultListView: View {
...
+                    Image(decorative: Asset.Images.plus)
+                        .resizable()
+                        .frame(width: 16, height: 16)
+                    Text(Localizations.addAnItem)
                 }
-                .padding(.horizontal, 16)
-                .frame(minHeight: reader.size.height)
             }
+            .buttonStyle(.primary(shouldFillWidth: false))
+            .clipShape(RoundedRectangle(cornerRadius: 28))
```
Comment (matt-livefront) [inline BitwardenShared/UI/Vault/Vault/VaultList/VaultListView.swift:105]: 🎨 I think we could leave the clipping out of this. There's going to be a ticket as part of the redesign to update all buttons to that new style and so we can build it into the button style at that point.

File: BitwardenShared/UI/Platform/Application/Support/Images.xcassets/Images/empty_vault_items.imageset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "images" : [
+    {
+      "filename" : "empty_vault_items_light.pdf",
```
Comment (matt-livefront) [inline BitwardenShared/UI/Platform/Application/Support/Images.xcassets/Images/empty_vault_items.imageset/Contents.json:4]: ❓ Does it make sense to name this the same as the illustrations name in Figma (`items`)? There's pros/cons to both, but I like the fact that it matches Figma and we could then import all illustrations up front and not have to do the mapping of changing the name to where it's used.

File: BitwardenShared/UI/Vault/Vault/VaultList/VaultListView.swift [hunk 4]
```diff
@@ -69,30 +69,44 @@ private struct SearchableVaultListView: View {
 
     /// A view that displays the empty vault interface.
     @ViewBuilder private var emptyVault: some View {
-        GeometryReader { reader in
-            ScrollView {
```
Comment (matt-livefront) [inline BitwardenShared/UI/Vault/Vault/VaultList/VaultListView.swift:73]: ⚠️ I think we still want the scroll view so it scrolls on smaller devices or with large fonts.

Screenshot 2024-10-08 at 9 35 12 AM

File: BitwardenShared/UI/Platform/Application/Support/Images.xcassets/Images/empty_vault_items.imageset/Contents.json (hunk 3 above)
Comment (phil-livefront) [reply BitwardenShared/UI/Platform/Application/Support/Images.xcassets/Images/empty_vault_items.imageset/Contents.json:4]: yeah, i can do that!

**Review by matt-livefront — APPROVED** [review]

**Review by fedemkr — APPROVED** [review]

**Pull request merged**